.github/workflows/auto-deploy-contracts.yaml

```diff
@@ -0,0 +1,85 @@
+name: Auto deploy IPC contracts when changed
+
+on:
+  workflow_dispatch:
+  # TODO: Test only. Remove before PR merge
+  pull_request:
+  push:
+    branches:
+      - main
+    paths:
+      - contracts/**
+
+concurrency:
+  # Only allow one run at a time for this workflow
+  group: auto-deploy-contracts
+  cancel-in-progress: true
+
+jobs:
+  deploy-contracts:
+    runs-on: ubuntu-latest
+    env:
+      RPC_URL: https://calibration.filfox.info/rpc/v1
+      PRIVATE_KEY: ${{ secrets.CONTRACTS_DEPLOYER_PRIVATE_KEY }}
+    steps:
+      - name: Checkout code repo
+        uses: actions/checkout@v4
+        with:
+          # TODO(jie): After switch to workflow_dispatch only, we should use ref_name.
+          # head_ref only works for workflow triggered by pull requests.
+          # ref: ${{ github.ref_name }}
+          ref: ${{ github.head_ref }}
+          submodules: recursive
+      - name: Setup node and npm
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'npm'
+          cache-dependency-path: contracts/package-lock.json
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+      - name: Deploy IPC contracts to calibration net
+        id: deploy_contracts
+        run: |
+          cd contracts
+          npm install --save hardhat
+          output=$(make deploy-ipc NETWORK=calibrationnet)
+          echo "deploy_output<<EOF" >> $GITHUB_OUTPUT
+          echo "$output" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+      - name: Parse deploy output
+        run: |
+          deploy_output='${{ steps.deploy_contracts.outputs.deploy_output }}'
+          echo "$deploy_output"
+          deployed_gateway_address=$(echo "$deploy_output" | grep '"Gateway"' | awk -F'"' '{print $4}')
+          deployed_registry_address=$(echo "$deploy_output" | grep '"SubnetRegistry"' | awk -F'"' '{print $4}')
+          echo "gateway_address=$deployed_gateway_address" >> $GITHUB_ENV
+          echo "registry_address=$deployed_registry_address" >> $GITHUB_ENV
+          echo "commit_hash=$(git rev-parse HEAD)" >> $GITHUB_ENV
+      - name: Review deployed addresses
+        run: |
+          echo "commit_hash: $commit_hash"
+          echo "gateway_address: $gateway_address"
+          echo "registry_address: $registry_address"
+      - name: Switch code repo to cd/contract branch
```

Comment:
@mb1896 we're missing the merge from `main` into the cd branch, prior to committing the file. Can you please add it? This will (a) place the source and the addresses in one convenient branch, (b) create a nice merge graph tracking at the git history level what got deployed when.

Reply:
Fixed. Please take another look.